File: assets/backing-services/typescript/tests/checkpoint-store-memory.test.ts
import { createInMemoryCheckpointStore } from '../../src/adapters/driven/checkpoint-store-memory.js';
import { createInMemoryEventStore } from '../../src/adapters/driven/event-store-memory.js';
import { checkpointStoreContract } from './checkpoint-store-contract.js';

/**
 * The checkpoint contract, against the fake. Runs in `make verify` with no Docker, which is the whole
 * reason the in-memory adapters exist. What it cannot prove is two workers genuinely racing for one
 * lease — nothing here runs concurrently — and `make test-integration` is where that is proved.
 */
checkpointStoreContract('the in-memory checkpoint store', async () => {
  const store = createInMemoryEventStore();
  return { store, checkpoints: createInMemoryCheckpointStore(store) };
});

File: assets/backing-services/java/tests/checkpoint_store_memory_test.java
package com.example.deliverystarter.adapters.driven.checkpointstorememory;

import com.example.deliverystarter.adapters.driven.eventstorememory.InMemoryEventStore;
import com.example.deliverystarter.checkpointstorecontract.CheckpointStoreContract;

/**
 * The checkpoint contract, against the fake.
 *
 * <p>Runs in {@code make verify} with no Docker, which is the whole reason the in-memory adapters exist.
 * What it cannot prove is two workers genuinely racing for one lease — this one serialises them — and
 * {@code make test-integration} is where that is proved.
 */
class InMemoryCheckpointStoreTest extends CheckpointStoreContract {

    @Override
    protected Stores stores() {
        InMemoryEventStore store = new InMemoryEventStore();
        return new Stores(store, new InMemoryCheckpointStore(store));
    }
}
